stellargraph/mapper/sampled_link_generators.py

```diff
@@ -214,18 +214,44 @@ def __init__(self, G, batch_size, num_samples, seed=None, name=None):
 
         self.head_node_types = self.schema.node_types * 2
 
-        # The sampler used to generate random samples of neighbours
-        self.sampler = SampledBreadthFirstWalk(G, graph_schema=self.schema, seed=seed)
+        self._graph = G
+        self._seed = seed
+        self.random = random.Random(seed)
+        self._samplers = dict()
+
+    def _sampler(self, batch_num):
+        """
+        Get the sampler for a particular batch number. Each batch number has an associated sampler
+        with its own random state, so that batches being fetched in parallel do not interfere with
+        each other's random states. The seed for each sampler is a combination of the Sequence
+        object's seed and the batch number. For its intended use in a Keras/TF workflow, if there
+        are N batches in an epoch, there will be N samplers created, each corresponding to a
+        particular ``batch_num``.
 
-    def sample_features(self, head_links):
+        Args:
+            batch_num (int): Batch number
+
+        Returns:
+            SampledBreadthFirstWalk object
+        """
+        if batch_num in self._samplers:
```

Comment:
It would be very slightly faster (and more pythonic) to EAFP rather than LYBL here (https://devblogs.microsoft.com/python/idiomatic-python-eafp-versus-lbyl/):

```python
try:
    return self._samplers[batch_num]
except KeyError:
    seed = ...
    self._samplers[batch_num] = ...
    return self._samplers[batch_num]